source/common/http/http2/codec_impl.cc

```diff
@@ -1995,15 +1995,15 @@ ConnectionImpl::ClientHttp2Options::ClientHttp2Options(
     : Http2Options(http2_options, max_headers_kb) {
   og_options_.perspective = http2::adapter::Perspective::kClient;
   og_options_.remote_max_concurrent_streams =
-      ::Envoy::Http2::Utility::OptionsLimits::DEFAULT_MAX_CONCURRENT_STREAMS;
+      ::Envoy::Http2::Utility::OptionsLimits::MAX_MAX_CONCURRENT_STREAMS;
```

Comment:
This is inconsistent with the updated documentation in api/envoy/config/core/v3/protocol.proto

Reply:
+1, something doesn't seem quite right, here.

Either this (and the nghttp2 limit) should use the new default, conditioned by the runtime feature, or it should be using the value from `http2_options`.

Reply:
This is used to set the  assumed initial value of the remote endpoint's max concurrent streams setting. The MAX_MAX_CONCURRENT_STREAMS ensure we didn't change the preivous behavior.